File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = []
    for i in range(len(number)):
        while k > 0 and answer and int(answer[-1]) < int(number[i]):
            answer.pop()
            k -= 1
        answer.append(number[i])
    return ''.join(answer[:len(number)-k])

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    const stack = [];
    
    for(let i = 0; i < number.length; i++){
        while(stack.length > 0 && k > 0 && stack[stack.length-1] < number[i]){
            stack.pop();
            k--;
        }
        stack.push(number[i]);
    }
    
    while(k > 0){
        stack.pop();
        k--; 
    }
    
    var answer = stack.join("");
    return answer;
}